pkg/runctl/controller.go

```diff
@@ -428,6 +462,11 @@ func (c *Controller) syncHandler(key string) error {
 		if err != nil {
 			return c.onGetRunError(ctx, pipelineRunAPIObj, pipelineRun, err, api.StateCleaning, api.ResultErrorInfra, "running failed")
 		}
+		if run == nil {
```

Comment:
See https://github.com/SAP/stewardci-core/pull/345/files#r1052087271... run cannot be nil.

Reply:
If task run cannot be found nil,nil is returned. See https://github.com/SAP/stewardci-core/blob/PR_tekton_retry/pkg/runctl/run_manager.go#L713.
This was changed in this PR.
This is to avoid nil pointer exception if the task run is deleted e.g. manually.